README.md
# Advance_Notification

Advance_Notification is a Flutter package with custom snackabar.


## Development is continue

- There are **many new features planned** for Advance_Notification.



# Installing

### 1. Depend on it
Add this to your package's `pubspec.yaml` file:

```yaml
dependencies:
  smart_notification: <latest_version>
```


### 2. Install it

You can install packages from the command line:

Run this command with flutter:


```yaml
flutter pub add smart_notification
```

### 3. Import it

Now in your `Dart` code, you can use:

```dart
import 'package:smart_notification/smart_notification.dart';
```

## Getting Started || Usage

### Why Advance Notification

The Advance_Notification provide multiple design of snackbar notification.
### Examples

Advance_Notification have three mode in snackbar =>  BASIC,ADVANCE,MODERN.
```diff
-  Note: 
- Veriosn 1.0.6 or above , we change iconposition, type, mode String type to enum type.
- So you can use like -> 
- iconPosition:IconPosition.LEFT or IconPosition.RIGHT  or IconPosition.CENTER AND
- type:sType.DARK or sType.ERROR or sType.WARNING or sType.PRIMARY or sType.INFO or sType.LIGHT or sType.SUCCESS  AND
- mode: Mode.BASIC or Mode.MODERN or Mode.ADVANCE
```

### 1: Basic Mode

<hr>

You can use basic snackbar with background color, text color, position

<img src="https://github.com/shyambhati/Advance_Notification/blob/main/display/basic.gif?raw=true" align = "right" width="300px" height = "600px">

#### 1.1 => Example
**Note:** For basic mode you don't need to write mode.
```dart
AdvanceSnackBar(message: "Hello...").show(context);
```


**Note:** Basic with background color.
```dart
 AdvanceSnackBar(
     message: "Notification Message", 
     bgColor: Colors.blueAccent).show(context);
```


#### 1.2 => Example
**Note:** Basic with background color or text color.

```dart
AdvanceSnackBar(
    message: "Notification Message",
    bgColor: Colors.tealAccent,
    textColor: Colors.red).show(context);
```


#### 1.3 => Example
**Note:** Basic with background color or text  or fixed position.
```dart
AdvanceSnackBar(
    message: "Notification Message",
    bgColor: Colors.tealAccent,
    textColor: Colors.red,
    isFixed: false).show(context);
```

### 2: Advance mode 

You can use advance mode snackbar with many features. It has default close button.
You can override the close button event, color and icon image. 

<img src="https://github.com/shyambhati/Advance_Notification/blob/main/display/advance.gif?raw=true" align = "right" width="300px" height = "600px">

#### 2.1 => Example

```dart
 AdvanceSnackBar(
     message: "Notification Message",
     mode: "ADVANCE",).show(context);
```

#### 2.2 => Example

You can add your custom duration for hide notification. default duration is 4 seconds. We have multiple types in Advance mode like : PRIMARY,SECONDARY,LIGHT,DARK,SUCCESS,INFO,WARNING,ERROR,default type is SUCCESS.

```dart

AdvanceSnackBar(
    message: "Notification Message",
    mode: "ADVANCE",
    duration: Duration(seconds: 5),).show(context);
```

#### 2.3 => Example
**Note:** Change bgColor, text color close iconColor.

```dart
AdvanceSnackBar(
    message: "Notification Message",
    mode: "ADVANCE",
    duration: Duration(seconds: 5),
    bgColor: Colors.red,
    textColor: Colors.black,
    iconColor: Colors.black,).show(context);
```

#### 2.4 => Example
**Note:** With this example you can change textSize or show notification icon.

```dart
AdvanceSnackBar(
        message: "Notification Message",
        mode: "ADVANCE",
        duration: Duration(seconds: 5),
        bgColor: Colors.red,
        textColor: Colors.white,
        iconColor: Colors.white,
        textSize: 22,
        isIcon: true,
    ).show(context);

```

#### 2.5 => Example
**Note:** Every type has own icon you can show and hide with isIcon.
```dart
AdvanceSnackBar(
        message: "Notification Message",
        mode: "ADVANCE",
        duration: Duration(seconds: 5),
        type: "ERROR",
        textSize: 20,
        isIcon: true,
    ).show(context);
```

#### 2.6 => Example
**Note:** Change type icon position.
```dart
AdvanceSnackBar(
        message: "Notification Message",
        mode: "ADVANCE",
        duration: Duration(seconds: 5),
        type: "ERROR",
        textSize: 20,
        isIcon: true,
        iconPosition: "RIGHT",
    ).show(context);
```

#### 2.7 => Example
**Note:** You can also add tittle.
```dart
AdvanceSnackBar(
        message: "Notification Message",
        tittle: "Tittle message",
        mode: "ADVANCE",
        type: "ERROR",
        textSize: 20,
        isIcon: true,
        closeIconPosition: "LEFT", 
    ).show(context);

```

#### 2.8 => Example
**Note:** You can change divider color (hr line).
```dart

AdvanceSnackBar(
        message: "Notification Message",
        tittle: "Tittle message",
        mode: "ADVANCE",
        type: "ERROR",
        textSize: 20,
        isIcon: true,
        dividerColor: Colors.red,
        closeIconPosition: "LEFT", 
    ).show(context);

```

#### 2.9 => Example
**Note:** With child you can add an extra widgets it can be a button, icon, image, etc.
```dart
AdvanceSnackBar(
        message: "Notification Message ",
        mode: "ADVANCE",
        type: "PRIMARY",
        tittle: "Tittle message",
        iconPosition: "RIGHT",
        child: Padding(
            padding: const EdgeInsets.only(left: 2),
            child: Icon(
            Icons.all_inbox,
            color: Colors.red,
            size: 25,
            ),
        ),
        isIcon: true)
    .show(context);
```

#### 2.10 => Example
**Note:** You can override onclick function of close button.

```dart
AdvanceSnackBar(
        message: "Notification Message ",
        mode: "ADVANCE",
        type: "PRIMARY",
        onClick: () 
        {
            print("Hello");
        },
        isIcon: true).show(context);

```

### 3: Modern mode 
Modern mode have all advance mode features and it have some new features. 
It has a unique design. 
You can change advance mode design with add a single word "MODERN".
In modern mode you can't change divider color we are working on it.

<img src="https://github.com/shyambhati/Advance_Notification/blob/main/display/modern.gif?raw=true" align = "right" width="300px" height = "600px">

#### 3.1 => Example
```dart
AdvanceSnackBar(
        message: "Notification Message",
        mode: "MODERN",).show(context);

```
#### 3.2 => Example

**Note:** You can change the type like advance mode.

```dart
AdvanceSnackBar(
        message: "Notification Message",
        mode: "MODERN",type: "WARNING")).show(context);
```

#### 3.3 => Example
```dart
AdvanceSnackBar(
        message: "Notification Message",
        mode: "MODERN",
        type: "ERROR",
        isIcon: true)
    .show(context);
```

#### 3.4 => Example
```dart
AdvanceSnackBar(
        message: "Notification Message content",
        mode: "MODERN",
        type: "ERROR",
        tittle: "Tittle message",
        isIcon: true)
    .show(context);
```
<hr>

### The component takes 1 compulsory prop - message. Other props are optional. The table below explains more.

| Prop              | Mode            |  Type     | Description/Purpose                | Default     | Required |
| :-----------------| :-------------: | :-------: | :--------------------------------: | :---------- |:------   |
| tittle            | Advance,Modern  |`String`   |It is for notification tittle.      | ""          |  NO      |
| message           | All             |`String`   |It is for notification Message.      | ""          |  YES     |
| duration          | Advance,Modern  |`Duration` |Duration of notification hide.         | "Duration(seconds: 4)"          |  NO      |
| isIcon            | Advance,Modern  |`String`   |Icon show in advance and modern.                  | false          |  NO      |
| isFixed           | Basic           |`bool`     |It will work in basic mode only.                   | ""          |  true      |
| isClosable        | Advance,Modern  |`bool`     |Hide/show close icon.                 | true          |  NO      |
| isDivider         | Advance,Modern  |`bool`     |It will show when you add tittle.                   | true          |  NO      |
| isChildLeft       | Advance,Modern  |`bool`   |It swap close icon and child widget left to right.                   | true          |  NO      |
| type              | Advance,Modern  |`String`   | These are default design of snackbar | PRIMARY, SECONDARY, LIGHT, DARK, SUCCESS, INFO, WARNING, ERROR.          |  NO      |
| mode              |  |`String`   |BASIC, ADVANCE, MODERN.                   | BASIC          |  NO      |
| iconPosition      | Advance,Modern  |`String`   |You can change icon position of types icons.     | "LEFT"          |  NO      |
| closeIconPosition | Advance,Modern  |`String`   |You can also change closeIconPosition.           | "RIGHT"          |  NO      |
| onClick           | Advance,Modern  |`function` |Override the closeButton onclick function.                   | ""          |  NO      |
| icon              | Advance,Modern  |`widget`   |You can change icon.                   | ""          |  NO      |
| child             | Advance,Modern  |`widget`   |You can add extra widget.                   | ""          |  NO      |
| borderRadius      | Advance,Modern  |`double`   |Can change borderRadius.             | 10          |  NO      |
| mHeight           | Advance,Modern  |`double`   |Can change snackbar height.                  | 40          |  NO      |


## Bugs or Requests
If you encounter any problems feel free to open an [issue](https://github.com/aagarwal1012/Animated-Text-Kit/issues/new?assignees=&labels=&template=bug_report.md). If you feel the library is missing a [feature](https://github.com/aagarwal1012/Animated-Text-Kit/issues/new?assignees=&labels=&template=feature_request.md), please raise a ticket on GitHub and I'll look into it. Pull request are also welcome.



CHANGELOG.md
## 1.0.6

-- Change iconposition, type, mode String type to enum type.

## 1.0.5

-- Change some documents details.

## 1.0.4

-- Change some documents details.

## 1.0.3

-- Change some documents details.

## 1.0.2

-- Change Some documents.

## 1.0.1

-- Initial Release.
-- There are **many new features planned** for Advance_Notification.


test/advance_notification_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:advance_notification/advance_notification.dart';

void main() {
  test('Simple message', () {
    AdvanceSnackBar(message: "message");
  });
}


lib/AdvanceSnackBar.dart
part of advance_notification;

class AdvanceSnackBar {
  final String tittle;
  final String message;

  final Duration duration;
  final TextAlign textAlign;

  final bool isFixed;
  final sType type;

  final Mode mode;
  final FontWeight fontWeight;
  final Color bgColor;
  final Color textColor;
  final Color iconColor;
  final Color dividerColor;
  final double tittleSize;
  final double textSize;
  final bool isIcon;
  final bool isDivider;
  final bool isClosable;
  final bool isChildLeft;

  final double iconSize;
  // final String iconPosition;
  final IconPosition iconPosition;
  final IconPosition closeIconPosition;
  final VoidCallback? onClick;
  final Widget? icon;
  final Widget? child;
  final double borderRadius;
  final double mHeight;
  final double margin;

  const AdvanceSnackBar({
    this.message = "Advance Snackbar",
    this.tittle = "",
    this.bgColor = const Color(0xFF323232),
    this.textColor = const Color(0xFFffffff),
    this.iconColor = const Color(0xFFffffff),
    this.dividerColor = Colors.white70,
    this.tittleSize = 18,
    this.textSize = 16,
    this.iconSize = 24.0,
    this.fontWeight = FontWeight.w500,
    this.textAlign = TextAlign.left,
    this.duration = const Duration(seconds: 4),
    this.isIcon = false,
    this.isFixed = true,
    this.isClosable = true,
    this.isDivider = true,
    this.isChildLeft = true,
    this.type = sType.SUCCESS,
    this.mode = Mode.BASIC,
    this.iconPosition = IconPosition.NORMAL,
    this.closeIconPosition = IconPosition.NORMAL,
    this.onClick,
    this.icon,
    this.child,
    this.borderRadius = 10,
    this.mHeight = 40,
    this.margin = 20,
  });

  show(
    BuildContext context,
  ) {
    ScaffoldMessenger.of(context)
        .showSnackBar(
          SnackBar(
            behavior: mode == Mode.BASIC
                ? (isFixed ? SnackBarBehavior.fixed : SnackBarBehavior.floating)
                : SnackBarBehavior.floating,
            padding: mode != Mode.BASIC ? new EdgeInsets.all(0) : null,
            duration: duration,
            backgroundColor: mode != Mode.BASIC ? Colors.transparent : bgColor,
            content: __genrateBar(context),
            elevation: 0,
          ),
        )
        .closed
        .then((value) => ScaffoldMessenger.of(context).clearSnackBars());
  }

  Widget __genrateBar(context) {
    return mode != Mode.BASIC
        ? Container(
            margin: new EdgeInsets.symmetric(horizontal: 5),
            child: ClipRRect(
              borderRadius: BorderRadius.all(
                  Radius.circular(mode == Mode.ADVANCE ? borderRadius : 0)),
              child: Container(
                decoration: BoxDecoration(
                  border: mode == Mode.MODERN
                      ? Border(
                          left: BorderSide(
                            width: 10.0,
                            color:
                                __generateBgColor(), //__getColor("SUCCESS", "dark"),
                          ),
                        )
                      : null,
                  color: mode == Mode.MODERN
                      ? lighten(__generateBgColor(), .2)
                      : __generateBgColor(),
                ),
                width: MediaQuery.of(context).size.width,
                padding: new EdgeInsets.all(5),
                height: mHeight > 40
                    ? mHeight
                    : tittle == ""
                        ? 40.0
                        : (tittleSize + textSize + 25),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.stretch,
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: <Widget>[
                    Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      textDirection: closeIconPosition == IconPosition.LEFT
                          ? TextDirection.rtl
                          : TextDirection.ltr,
                      children: [
                        Expanded(
                          child: Container(
                            child: Row(
                              textDirection: iconPosition == IconPosition.RIGHT
                                  ? TextDirection.rtl
                                  : TextDirection.ltr,
                              mainAxisAlignment: __generatemainAxisAlignment(),
                              children: [
                                isIcon
                                    ? Container(
                                        padding: new EdgeInsets.symmetric(
                                            horizontal: 5),
                                        child: __genrateIcon(),
                                      )
                                    : SizedBox.shrink(),
                                __genrateText(),
                              ],
                            ),
                          ),
                        ),
                        (isClosable || child != null)
                            ? Row(
                                textDirection: isChildLeft
                                    ? __generateRtl(true)
                                    : __generateRtl(false),
                                // isChildLeft?__generateRtl(true):,
                                children: [
                                  isClosable
                                      ? GestureDetector(
                                          onTap: () {
                                            onClick == null
                                                ? ScaffoldMessenger.of(context)
                                                    .hideCurrentSnackBar()
                                                : onClick!.call();
                                          },
                                          child: Container(
                                            margin: new EdgeInsets.symmetric(
                                                horizontal: 5),
                                            child: __closeIcon(),
                                          ),
                                        )
                                      : SizedBox.shrink(),
                                  child != null
                                      ? Container(child: child)
                                      : SizedBox.shrink(),
                                ],
                              )
                            : SizedBox.shrink(),
                      ],
                    ),
                  ],
                ),
              ),
            ),
          )
        : __genrateText();
  }

  __genrateIcon() {
    var icon = Icons.check_circle;

    if (type == sType.ERROR)
      icon = Icons.error;
    else if (type == sType.WARNING)
      icon = Icons.warning_rounded;
    else if (type == sType.INFO) icon = Icons.info;
    return Icon(
      icon,
      color: mode == Mode.MODERN
          ? __generateBgColor()
          : __generateColor(1.0, iconColor),
      size: tittle == "" ? iconSize : iconSize + 10,
    );
  }

  __closeIcon() {
    return icon != null
        ? icon
        : Icon(
            Icons.close_outlined,
            color: __generateColor(1.0, iconColor),
            size: tittle == "" ? iconSize : iconSize + 5,
          );
  }

  __generateBgColor() {
    String c = Color(0xFF323232).toString();
    if (bgColor.toString() != c)
      return bgColor;
    else if (type == sType.DARK)
      c = "161c2d";
    else if (type == sType.ERROR)
      c = "df4759";
    else if (type == sType.WARNING)
      c = "ffc107";
    else if (type == sType.INFO)
      c = "467fd0";
    else if (type == sType.PRIMARY)
      c = "7c69ef";
    else if (type == sType.SECONDARY)
      c = "d9e2ef";
    else if (type == sType.LIGHT)
      c = "f1f4f8";
    else if (type == sType.SUCCESS)
      c = "42ba96";
    else
      c = "ffffff";
    return Color(int.parse("0xff$c"));
  }

  __generateColor(var opacity, var color) {
    if (type == sType.DARK) {
      return Colors.white.withOpacity(opacity);
    } else if (type == sType.SECONDARY ||
        type == sType.LIGHT ||
        mode == Mode.MODERN ||
        type != sType.DARK) {
      return Colors.black.withOpacity(opacity);
    } else {
      return color.withOpacity(opacity);
    }
  }

  __genrateText() {
    return (mode == Mode.BASIC || tittle == "")
        ? Padding(
            padding: const EdgeInsets.symmetric(horizontal: 5),
            child: __getText(message, textSize, FontWeight.normal, 1.0),
          )
        : Expanded(
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                __getText(tittle, tittleSize, FontWeight.bold, 1.0),
                isDivider
                    ? Divider(
                        color: mode == Mode.MODERN
                            ? lighten(Color(int.parse("0xff111111")), .1)
                            : dividerColor,
                        height: 2,
                        thickness: 1.5,
                      )
                    : SizedBox.shrink(),
                __getText(message, textSize, FontWeight.normal, 0.7),
              ],
            ),
          );
  }

  __getText(String text, double size, var fontWeight, var opacity) {
    return Text(
      text,
      overflow: TextOverflow.ellipsis,
      style: TextStyle(
          color: mode == Mode.MODERN
              ? __generateColor(opacity, textColor)
              : textColor,
          fontWeight: fontWeight,
          fontSize: size),
    );
  }

  __generatemainAxisAlignment() {
    if (isClosable || iconPosition == IconPosition.NORMAL)
      return MainAxisAlignment.start;
    else if (iconPosition == IconPosition.LEFT ||
        iconPosition == IconPosition.RIGHT)
      return MainAxisAlignment.spaceBetween;
    else if (iconPosition == IconPosition.CENTER)
      return MainAxisAlignment.center;
  }

  __generateRtl(var type) {
    return type ? TextDirection.rtl : TextDirection.ltr;
  }

  Color darken(Color color, [double amount = .1]) {
    assert(amount >= 0 && amount <= 1);
    final hsl = HSLColor.fromColor(color);
    final hslDark = hsl.withLightness((hsl.lightness - amount).clamp(0.0, 1.0));
    return hslDark.toColor();
  }

  Color lighten(Color color, [double amount = .1]) {
    assert(amount >= 0 && amount <= 1);
    final hsl = HSLColor.fromColor(color);
    final hslLight =
        hsl.withLightness((hsl.lightness + amount).clamp(0.0, 1.0));
    return hslLight.toColor();
  }
}

enum IconPosition { NORMAL, LEFT, RIGHT, CENTER }
enum sType { DARK, ERROR, WARNING, INFO, PRIMARY, SECONDARY, LIGHT, SUCCESS }
enum Mode { BASIC, MODERN, ADVANCE }


Example/lib/main.dart
import 'package:advance_notification/advance_notification.dart';
import 'package:flutter/material.dart';

class NotificationDemo extends StatefulWidget {
  const NotificationDemo({Key? key}) : super(key: key);

  @override
  _NotificationDemoState createState() => _NotificationDemoState();
}

class _NotificationDemoState extends State<NotificationDemo> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Container(
        child: SafeArea(
            child: Center(
          child: IntrinsicWidth(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.stretch,
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Ebutton(
                  text: "Notification | Basic ",
                  color: Colors.purple,
                  onClick: () {
                    AdvanceSnackBar(message: "Hello....").show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Basic 1",
                  color: Colors.purple,
                  onClick: () {
                    AdvanceSnackBar(
                            message: "Notification Message",
                            bgColor: Colors.blueAccent)
                        .show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Basic 2",
                  color: Colors.purple,
                  onClick: () {
                    AdvanceSnackBar(
                            message: "Notification Message",
                            bgColor: Colors.tealAccent,
                            textColor: Colors.red)
                        .show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Basic 3",
                  color: Colors.purple,
                  onClick: () {
                    AdvanceSnackBar(
                            message: "Notification Message",
                            bgColor: Colors.blueAccent,
                            textColor: Colors.white,
                            isFixed: false)
                        .show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Advance 1",
                  color: Colors.redAccent,
                  onClick: () {
                    AdvanceSnackBar(
                      message: "Notification Message",
                      mode: Mode.ADVANCE,
                    ).show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Advance 2",
                  color: Colors.redAccent,
                  onClick: () {
                    AdvanceSnackBar(
                      message: "Notification Message",
                      mode: Mode.ADVANCE,
                      duration: Duration(seconds: 5),
                      bgColor: Colors.red,
                      textColor: Colors.black,
                      iconColor: Colors.black,
                    ).show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Advance 3",
                  color: Colors.redAccent,
                  onClick: () {
                    AdvanceSnackBar(
                      message: "Notification Message",
                      mode: Mode.ADVANCE,
                      duration: Duration(seconds: 5),
                      bgColor: Colors.red,
                      textColor: Colors.white,
                      iconColor: Colors.white,
                      textSize: 22,
                      isIcon: true,
                    ).show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Advance 4",
                  color: Colors.redAccent,
                  onClick: () {
                    AdvanceSnackBar(
                      message: "Notification Message",
                      mode: Mode.ADVANCE,
                      duration: Duration(seconds: 5),
                      textColor: Colors.white,
                      iconColor: Colors.white,
                      type: sType.ERROR,
                      textSize: 20,
                      isIcon: true,
                      iconPosition:
                          IconPosition.RIGHT, // it shold be after text
                    ).show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Advance 5",
                  color: Colors.redAccent,
                  onClick: () {
                    AdvanceSnackBar(
                      message: "Notification Message",
                      mode: Mode.ADVANCE,
                      duration: Duration(seconds: 5),
                      textColor: Colors.white,
                      iconColor: Colors.white,
                      type: sType.ERROR,
                      textSize: 20,
                      isIcon: true,
                      closeIconPosition: IconPosition.LEFT,
                      iconPosition:
                          IconPosition.RIGHT, // it shold be after text
                    ).show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Modern 1",
                  color: Colors.brown,
                  onClick: () {
                    AdvanceSnackBar(
                            message: "Notification Message", mode: Mode.MODERN)
                        .show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Modern 2",
                  color: Colors.brown,
                  onClick: () {
                    AdvanceSnackBar(
                            message: "Notification Message",
                            mode: Mode.MODERN,
                            type: sType.WARNING)
                        .show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Modern 3",
                  color: Colors.brown,
                  onClick: () {
                    AdvanceSnackBar(
                            message: "Notification Message",
                            mode: Mode.MODERN,
                            type: sType.ERROR,
                            isIcon: true)
                        .show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Modern 4",
                  color: Colors.brown,
                  onClick: () {
                    AdvanceSnackBar(
                            message: "Notification Message content",
                            mode: Mode.MODERN,
                            type: sType.ERROR,
                            tittle: "Tittle message",
                            isIcon: true)
                        .show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Modern 5",
                  color: Colors.brown,
                  onClick: () {
                    AdvanceSnackBar(
                            message: "Notification Message ",
                            mode: Mode.MODERN,
                            type: sType.SUCCESS,
                            tittle: "Tittle message",
                            iconPosition: IconPosition.RIGHT,
                            child: Padding(
                              padding: const EdgeInsets.only(left: 2),
                              child: Icon(
                                Icons.all_inbox,
                                color: Colors.red,
                                size: 25,
                              ),
                            ),
                            isIcon: true)
                        .show(context);
                  },
                ),
                Ebutton(
                  text: "Notification | Modern 6",
                  color: Colors.brown,
                  onClick: () {
                    AdvanceSnackBar(
                            message: "Notification Message ",
                            mode: Mode.ADVANCE,
                            type: sType.PRIMARY,
                            tittle: "Tittle message",
                            iconPosition: IconPosition.RIGHT,
                            dividerColor: Colors.red,
                            onClick: () {
                              print("Hello");
                            },
                            isIcon: true)
                        .show(context);
                  },
                ),
              ],
            ),
          ),
        )),
      ),
    );
  }
}

/// Ebutton is a custom ElevatedButton widget
class Ebutton extends StatelessWidget {
  final String text;
  final Color color;
  final VoidCallback? onClick;

  const Ebutton({
    @required this.text = "",
    this.color = Colors.blueAccent,
    this.onClick,
  });

  @override
  Widget build(BuildContext context) {
    return ElevatedButton(
      onPressed: () {
        onClick!.call();
      },
      child: Text(text),
      style: ElevatedButton.styleFrom(
        primary: color, //
      ),
    );
  }
}
